pkg/loki/convert.go

```diff
@@ -24,6 +24,7 @@ func (q *Query) convertToAnyMatch() *Query {
 	}
 
 	out := Query{
+		baseURL:      q.baseURL,
```

Comment:
this one was really tricky to spot, resulting in bug for the "match any" case. I'd like to improve code safety here ... maybe should use explicit constructors instead